src/io/flutter/devtools/DevToolsManager.java

```diff
@@ -298,6 +242,36 @@ private void openBrowserImpl(String uri, String screen) {
     }
   }
 
+  private CompletableFuture<DevToolsInstance> getDevToolsInstance() {
+    final CompletableFuture<DevToolsInstance> instance = new CompletableFuture<>();
+    final Optional<FlutterApp> appsOptional =
+      FlutterApp.allFromProjectProcess(project).stream().filter((FlutterApp app) -> app.getProject() == project).findFirst();
```

Comment:
discussing offline if there is a more robust way to achieve the same result. I worry about all the extra edge cases this could introduce.

Reply:
I tried passing an app through to this function, but it's kind of complicated. There are a couple issues:
- App has to be found this way when opening devtools from the run/debug tab, and this function is used from that workflow
- App doesn't exist when a test is running, but I think we still want to be able to open devtools for bazel projects during tests? I'm not sure about the best way to make requests to the daemon from a test.